frontend/src/components/submission/SubmissionStatistics.vue

```diff
@@ -0,0 +1,222 @@
+<script setup lang="ts">
+import { version as uuidVersion, validate as uuidValidate } from 'uuid';
+import { onMounted, ref, watch } from 'vue';
+
+import { Calendar, Dropdown } from '@/lib/primevue';
+import { chefsService, userService } from '@/services';
+import { Regex } from '@/utils/constants';
+
+import type { Ref } from 'vue';
+import type { IInputEvent } from '@/interfaces';
+import type { User } from '@/types';
+
+// Types
+type StatisticFilters = {
+  dateFrom?: string;
+  dateTo?: string;
+  monthYear?: string;
+  userId?: string;
+};
+
+// State
+const loading: Ref<boolean> = ref(false);
+const statistics: Ref<any> = ref(undefined);
+const assigneeOptions: Ref<Array<User>> = ref([]);
+const statisticFilters: Ref<StatisticFilters> = ref({});
+
+// Actions
+const getAssigneeOptionLabel = (e: User) => `${e.fullName} [${e.email}]`;
+
+const getPercentage = (input: number) =>
+  statistics.value.total_submissions > 0 ? (input / statistics.value.total_submissions) * 100 : 0;
+
+const isEmpty = (value: unknown) =>
+  value === null || value === undefined || (typeof value === 'string' && value.trim().length === 0);
+
+async function onAssigneeInput(e: IInputEvent) {
+  const input = e.target.value;
+
+  if (input.length >= 3) {
+    assigneeOptions.value = (await userService.searchUsers({ email: input, fullName: input, username: input })).data;
+  } else if (input.match(Regex.EMAIL)) {
+    assigneeOptions.value = (await userService.searchUsers({ email: input })).data;
+  } else {
+    assigneeOptions.value = [];
+  }
+}
+
+watch(
+  statisticFilters,
+  async () => {
+    let valid =
+      isEmpty(statisticFilters.value.userId) ||
+      (!isEmpty(statisticFilters.value.userId) &&
+        uuidValidate(statisticFilters.value.userId as string) &&
+        uuidVersion(statisticFilters.value.userId as string) === 4);
+
+    if (valid) {
```

Comment:
i wonder if you need this valid check.
maybe userId will either be empty or a uuid.. both work in the query.(?)

Reply:
userId is a uuid once a user has been selected from the drop down. Up until that point it is filled with whatever the user is typing into the box. Kind of annoying how it works.